unitstatushandler/unitstatushandler.go

```diff
@@ -318,6 +325,22 @@ func (instance *Instance) StartSOTAUpdate() (err error) {
 	return instance.softwareManager.startUpdate()
 }
 
+// CloudConnected indicates unit connected to cloud.
+func (instance *Instance) CloudConnected() {
+	instance.connectionMutex.Lock()
+	defer instance.connectionMutex.Unlock()
+
+	instance.isConnected = true
+}
+
+// CloudDisconnected indicates unit disconnected from cloud.
+func (instance *Instance) CloudDisconnected() {
+	instance.connectionMutex.Lock()
+	defer instance.connectionMutex.Unlock()
```

Comment:
if we use lock only for `isConnected` variable then we could make it atomic and in this case, we don't need to create a mutex